uk.co.compucorp.civicrm.hrcore/CRM/HRCore/Form/CreateUserRecordTaskForm.php

```diff
@@ -0,0 +1,213 @@
+<?php
+
+use CRM_Utils_Array as ArrayHelper;
+use CRM_HRCore_Service_DrupalUserService as DrupalUserService;
+
+class CRM_HRCore_Form_CreateUserRecordTaskForm extends CRM_Contact_Form_Task {
+
+  /**
+   * @var array
+   */
+  protected $contactDetails = [];
+
+  /**
+   * @var bool
+   */
+  protected $sendEmail = FALSE;
+
+  /**
+   * @var DrupalUserService
+   */
+  protected $drupalUserService;
+
+  /**
+   * @param null $state
+   * @param mixed $action
+   * @param string $method
+   * @param null $name
+   */
+  public function __construct(
+    $state = NULL,
+    $action = CRM_Core_Action::NONE,
+    $method = 'post',
+    $name = NULL
+  ) {
+    $this->drupalUserService = Civi::container()->get('drupal_user_service');
+    parent::__construct($state, $action, $method, $name);
+  }
+
+  /**
+   * Build the form object.
+   */
+  public function buildQuickForm() {
+    CRM_Utils_System::setTitle(ts('Create User Records'));
+    $this->addDefaultButtons(ts('Create Records'));
+    $this->add('advcheckbox', 'sendEmail', ts('Send Email'));
+  }
+
+  /**
+   * Fetch contact details and set some template variables
+   */
+  public function preProcess() {
+    parent::preProcess();
+    $this->initContactDetails();
+    $haveNoAccount = $this->getContactsWithout('uf_id');
+    $haveAccount = array_diff_key($this->contactDetails, $haveNoAccount);
+    $this->assign('contactsWithoutEmail', $this->getContactsWithout('email'));
+    $this->assign('contactsWithAccount', $haveAccount);
+    $this->assign('contactsForCreation', $this->getValidContactsForCreation());
+    $this->assign('emailConflictContact', $this->getEmailConflictContacts());
+  }
+
+  /**
+   * Process the form after the input has been submitted and validated.
+   */
+  public function postProcess() {
+    $this->sendEmail = (bool) $this->getElementValue('sendEmail');
+    $contactsToCreate = $this->getValidContactsForCreation();
+
+    foreach ($contactsToCreate as $contact) {
+      $this->createAccount($contact);
+    }
+
+    CRM_Core_Session::setStatus(
+      ts('%1 new accounts were created', [1 => count($contactsToCreate)]),
+      ts('Updates Saved'),
+      'success'
+    );
+  }
+
+  /**
+   * Returns contacts that have an email and no drupal account
+   *
+   * @return array
+   */
+  private function getValidContactsForCreation() {
+    $missingEmail = $this->getContactsWithout('email');
```

Comment:
I agree the format was a bit different but thought it makes it nicely readable when you're looking at the method in use since it reads like a sentence, "get contacts without email". I can change it, but I like using "without" since I think it's more natural than "with empty .."